conan/cli/commands/create.py

```diff
@@ -168,3 +177,84 @@ def _get_test_conanfile_path(tf, conanfile_path):
         return test_conanfile_path
     elif tf:
         raise ConanException(f"test folder '{tf}' not available, or it doesn't have a conanfile.py")
+
+
+def _docker_runner(conan_api, profile, args, raw_args):
+    """
+    run conan inside a Docker continer
+    """
+    # import docker only if needed
+    import docker
+    docker_client = docker.from_env()
+    docker_api = docker.APIClient()
+    dockerfile = str(profile.remote.get('dockerfile', ''))
+    image = str(profile.remote.get('image', 'conanremote'))
+
+    remote_home = os.path.join(args.path, '.conanremote')
+    tgz_path = os.path.join(remote_home, 'conan_cache_save.tgz')
+    volumes = {
+        args.path: {'bind': args.path, 'mode': 'rw'}
+    }
+
+    environment = {
+        'CONAN_REMOTE_WS': args.path,
+        'CONAN_REMOTE_COMMAND': ' '.join(['conan create'] + raw_args),
+        'CONAN_REMOTE_ENVIRONMNET': '1'
+    }
+    # https://docker-py.readthedocs.io/en/stable/api.html#module-docker.api.build
+
+    ConanOutput().info(msg=f'\nBuilding the Docker image: {image}')
+    docker_build_logs = None
+    if dockerfile:
+        docker_build_logs = docker_api.build(path=dockerfile, tag=image)
+    else:
+        dockerfile = '''
+FROM ubuntu
+RUN apt update && apt upgrade -y
+RUN apt install -y build-essential
+RUN apt install -y python3-pip cmake git
+RUN cd /root && git clone https://github.com/davidsanfal/conan.git conan-io
+RUN cd /root/conan-io && pip install -e .
+'''
```

Comment:
Even if convenient, better remove this, use always an external input for docker.